src/main/java/io/aiven/kafka/connect/http/sender/AccessTokenHttpRequestBuilder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2021 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import java.net.URLEncoder;
+import java.net.http.HttpRequest;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+import io.aiven.kafka.connect.http.config.OAuth2AuthorizationMode;
+
+class AccessTokenHttpRequestBuilder implements HttpRequestBuilder {
+
+    private AccessTokenHttpRequestBuilder() {
+
+    }
+
+    @Override
+    public HttpRequest.Builder build(final HttpSinkConfig config) {
+        Objects.requireNonNull(config, "config");
+        Objects.requireNonNull(config.oauth2AccessTokenUri(), "oauth2AccessTokenUri");
+        final var accessTokenRequestBuilder = HttpRequest
+                        .newBuilder(config.oauth2AccessTokenUri())
+                        .timeout(REQUEST_HTTP_TIMEOUT)
+                        .header(HEADER_CONTENT_TYPE, "application/x-www-form-urlencoded");
+
+        final var accessTokenRequestBodyBuilder = new StringJoiner("&");
+        accessTokenRequestBodyBuilder.add(encodeNameAndValue("grant_type", "client_credentials"));
+        if (config.oauth2ClientScope() != null) {
+            accessTokenRequestBodyBuilder.add(encodeNameAndValue("scope", config.oauth2ClientScope()));
+        }
+
+        setClientIdAndSecret(accessTokenRequestBuilder, accessTokenRequestBodyBuilder, config);
+        return accessTokenRequestBuilder
+                .POST(HttpRequest.BodyPublishers.ofString(accessTokenRequestBodyBuilder.toString()));
+    }
+
+    private void setClientIdAndSecret(final HttpRequest.Builder httpRequestBuilder,
+                                      final StringJoiner requestBodyBuilder,
+                                      final HttpSinkConfig config) {
+        if (config.oauth2AuthorizationMode() == OAuth2AuthorizationMode.HEADER) {
+            final var clientAndSecretBytes =
+                    (config.oauth2ClientId() + ":" + config.oauth2ClientSecret().value())
+                            .getBytes(StandardCharsets.UTF_8);
+            final var clientAndSecretAuthHeader =
+                    "Basic " + Base64.getEncoder().encodeToString(clientAndSecretBytes);
+            httpRequestBuilder.header(HEADER_AUTHORIZATION, clientAndSecretAuthHeader);
+        } else {
```

Comment:
Lets explicitly specify that config.oauth2AuthorizationMode() == OAuth2AuthorizationMode.URL. It make intention a bit more clear